### ISSUE TITLE
LspInfo reports "Running in single file mode" on Windows after failing to match the root folder due to case sensitive comparison

### ISSUE THREAD

**Issue by spikedom** [neovim/nvim-lspconfig#2482]
### Description

LspInfo can incorrectly report "Running in single file mode" on Windows because it uses a case sensitive comparison in make_client_info when trying to match fname against candidate root directories.

The fname and candidate work_space folders may use different case.

In lspinfo.lua:
```
local function make_client_info(client, fname)

      for i = 1, #root_parts do
        -- Case sensitive comparison may fail to match equivalent path components on Windows
        -- if root_parts[i] ~= fname_parts[i] then

        -- Workaround hack:
        if vim.stricmp(root_parts[i],fname_parts[i]) ~= 0 then
          matched = false
          break
        end
      end
```

### Neovim version

NVIM v0.8.0-1210-gd367ed9b2
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3

### Nvim-lspconfig version

e20d77deac40e6e65d036acd60aff474c0f09282

### Operating system and version

Windows 10

### Affected language servers

all (observed with clangd but code appears to be common)

### Steps to reproduce

I'm afraid I don't know the commit hash of the version of lspinfo.lua that I have. Is the installed hash recorded somewhere?
The code in the the latest (hash as above) looks very similar.

I have so far failed to repro starting from the minimal config.

Editing a file by specifying a lowercase path can cause the LspInfo UI to fail to match the root dir:
:e c:\foo\bar\bar.cpp
:LspInfo

LspInfo reports "Running in single file mode".

Editing a file by specifying a path with the correct case (here the drive letter) works:
:e C:\foo\bar\bar.cpp
:LspInfo

LspInfo reports root directory is "C:/foo".

### Actual behavior

LspInfo reports "Running in single file mode".

### Expected behavior

LspInfo should report the root dir.

### Minimal config

```Lua
I have so far failed to repro starting from the minimal config.
```

### LSP log

Not available - the failure is in the LspInfo UI

**Comment by spikedom** [comment]
In the example the expected root directory, containing a .clangd file is C:\foo

**Comment by spikedom** [comment]
Apologies, this appears to be fixed by e20d77deac40e6e65d036acd60aff474c0f09282.
The minimal config was getting this newer version which was why I was unable to repro.

```
fname = uv.fs_realpath(fname) or fn.fnamemodify(fn.resolve(fname), ':p')
```
